src/apply.jl

```diff
@@ -25,16 +29,52 @@ function ITensors.apply(
     if ortho
       ψ = orthogonalize(ψ, v⃗[1])
     end
-    oψᵥ = apply(o, ψ[v⃗[1]] * ψ[v⃗[2]])
-    ψᵥ₁, ψᵥ₂ = factorize(
-      oψᵥ, inds(ψ[v⃗[1]]); cutoff, maxdim, tags=ITensorNetworks.edge_tag(e)
-    )
+
+    outer_dim_v1, outer_dim_v2 = dim(uniqueinds(ψ[v⃗[1]], o, ψ[v⃗[2]])),
+    dim(uniqueinds(ψ[v⃗[2]], o, ψ[v⃗[1]]))
+    dim_shared = dim(commoninds(ψ[v⃗[1]], ψ[v⃗[2]]))
+    d1, d2 = dim(commoninds(ψ[v⃗[1]], o)), dim(commoninds(ψ[v⃗[2]], o))
+    if outer_dim_v1 * outer_dim_v2 <= dim_shared * dim_shared * d1 * d2
+      Qᵥ₁, Rᵥ₁ = ITensor(1.0), copy(ψ[v⃗[1]])
```

Comment:
If we use `ITensor(true)` here it will make it work better with arbitrary element types (`ITensor(1.0)` is implicitly `Float64`, which will then force everything to be double precision).